### PR TITLE
Mixed callstack line numbers and multiple domain support

### PR DESCRIPTION
**Pull request by UnityAlex**
- Should this pull request have release notes?
  - [X] Yes
  - [ ] No
- Do these changes need to be back ported?
  - [ ] Yes
  - [X] No
- Do these changes need to be upstreamed to [mono/mono](https://github.com/mono/mono) or [dotnet/runtime](https://github.com/dotnet/runtime) repositories?
  - [ ] Yes
  - [X] No

Reviewers: please consider these questions as well! :heart:

**Release notes**

Internal VM-733 @UnityAlex :
Mono: Added multiple domain and line number support for the mixed callstack plugin.

### CODE REVIEW

File: mono/mini/mixed_callstack_plugin.c [hunk 1]
```diff
@@ -39,24 +81,40 @@ create_next_pmip_file()
...
 	MonoProfilerHandle prof = mono_profiler_create(NULL);
 	mono_profiler_set_domain_unloaded_callback(prof, mixed_callstack_plugin_on_domain_unload_end);
 
-	create_next_pmip_file();
+	// TODO: Clean this thing up somewhere on close
+	domainHandles = g_ptr_array_new();
+
+	// 1 is legacy 2 is file per domain
+	legacyMode = options == 1;
+	mono_get_root_domain();
```
Comment (joncham) [inline mono/mini/mixed_callstack_plugin.c]: Should this line be deleted?

File: mono/mini/mixed_callstack_plugin.c [hunk 2]
```diff
@@ -1,29 +1,71 @@
...
 
 	mixed_callstack_plugin_lock ();
 
+	HANDLE fileHandle = NULL;
+	DomainHandle* dHandle = get_domain_handle(domain_id);
+
+	if (dHandle)
+	{
+		fileHandle = dHandle->fileHandle;
+		pmipFileNum = ++dHandle->count;
```
Comment (joncham) [inline mono/mini/mixed_callstack_plugin.c:58]: Why do we need this? Because domain_id's get reused?

**Review by joncham — APPROVED** [review]

File: mono/mini/mixed_callstack_plugin.c (hunk 2 above)
Comment (UnityAlex) [reply mono/mini/mixed_callstack_plugin.c:58]: Yes I believe that was the original reason why we have a count value in the filename. It was there previously and I kept it. It makes it a little easier to be sure when debugging on the other side that you're dealing with a new domain.

File: mono/mini/mixed_callstack_plugin.c (hunk 1 above)
Comment (UnityAlex) [reply mono/mini/mixed_callstack_plugin.c]: oh yeah whoops thanks.

**Pull request merged**